backend/index.js

```diff
@@ -0,0 +1,81 @@
+const express = require('express');
+const knex = require('knex');
+const knexConfig = require('./knexfile');
+const db = knex(knexConfig.development);
+const server = express();
+const cors = require('cors');
+const morgan = require('morgan');
+const helmet = require('helmet');
+
+server.use(helmet());
+server.use(morgan('short'));
+server.use(cors());
+server.use(express.json());
+
+// server check
+server.get('/', (req, res) => {
+	res.send(`Api running on port: ${port}`);
+});
+
+// get all notes
+server.get('/api/notes', (req, res) => {
+	db('notes')
+		.then((note) => {
+			res.status(200).json(note);
+		})
+		.catch((err) => {
+			res.status(500).json(err);
+		});
+});
+
+// get a note by id
+server.get('/api/note/:id', (req, res) => {
+	const { id } = req.params;
+	db('notes')
+		.where({ id })
+		.then((note) => {
+			res.status(200).json(note);
+		})
+		.catch((err) => {
+			res.status(500).json(err);
+		});
+});
+
+// create a note
+server.post('/api/createnote', (req, res) => {
+	const note = req.body;
+	db('notes').insert(note).then((ids) => {
+		res.status(201).json(ids).catch((err) => {
```

Comment:
You need to place .catch in separete block or it will throw unhandled rejection on server side.